src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/ArrayMarshaller.cs

```diff
@@ -82,7 +82,7 @@ private IEnumerable<StatementSyntax> GeneratePinningPath(TypePositionInfo info,
         {
             (string managedIdentifer, string nativeIdentifier) = context.GetIdentifiers(info);
             string byRefIdentifier = $"__byref_{managedIdentifer}";
-            TypeSyntax arrayElementType = _elementType;
+            TypeSyntax arrayElementType = _elementType is PointerTypeSyntax ? PredefinedType(Token(SyntaxKind.ByteKeyword)) : _elementType;
```

Comment:
Honestly, I was just matching the return type of `GetArrayDataReference` because I didn't want to have to do all the casting. This was the (broken) code we were generating.
```csharp
ref int* __byref_arr = ref (arr == null ? ref *(int**)0 : ref System.Runtime.InteropServices.MemoryMarshal.GetArrayDataReference(arr));
fixed (byte* __arr_gen_native = &System.Runtime.CompilerServices.Unsafe.As<int*, byte>(ref __byref_arr))
```

This changes the `int*` to `byte`. They will still always be refs, so the `byte` is more of a lowest-common-denominator placeholder, not a 'we are casting the pointer element to a byte' (I can add a comment). Are you still concerned about the conversion here?

Reply:
Nope. I thought this code was converting a reference to the first element of a `int[]` to a `byte`. But this is really converting it to a `ref byte`, which is fine.